src/client_protocol/server.hpp

```diff
@@ -148,6 +151,8 @@ class query_server_t : public http_app_t {
     scoped_ptr_t<tcp_listener_t> tcp_listener;
 
     int next_thread;
+
+    SSL_CTX *tls_ctx;
```

Comment:
Though it's just a pointer and it probably won't matter in practice, this field should be declared above the `drainer` (e.g. below the `handler`). That ensures that during destruction, the `tls_ctx` field remains valid as long as connections can still be active or get established.